finetune.go

```diff
@@ -0,0 +1,226 @@
+package chatgpt
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+type FineTuningJobStatus string
+
+const (
+	FINETUNINGJOBSTATUS_CREATED   FineTuningJobStatus = "created"
+	FINETUNINGJOBSTATUS_PENDING   FineTuningJobStatus = "pending"
+	FINETUNINGJOBSTATUS_RUNNING   FineTuningJobStatus = "running"
+	FINETUNINGJOBSTATUS_SUCCEEDED FineTuningJobStatus = "succeeded"
+	FINETUNINGJOBSTATUS_FAILED    FineTuningJobStatus = "failed"
+	FINETUNINGJOBSTATUS_CANCELLED FineTuningJobStatus = "cancelled"
+)
+
+type FineTuningJob struct {
+	Object          string              `json:"object"`
+	Id              string              `json:"id"`
+	Model           string              `json:"model,omitempty"`
+	CreatedAt       int                 `json:"created_at"`
+	FinishedAt      int                 `json:"finished_at,omitempty"`
+	FineTunedModel  string              `json:"fine_tuned_model,omitempty"`
+	OrganizationId  string              `json:"organization_id,omitempty"`
+	ResultFiles     []string            `json:"result_files,omitempty"`
+	Status          FineTuningJobStatus `json:"status,omitempty"`
+	ValidationFile  interface{}         `json:"validation_file,omitempty"`
+	TrainingFile    string              `json:"training_file,omitempty"`
+	Hyperparameters struct {
+		NEpochs int `json:"n_epochs"`
```

Comment:
This can either be a string or an integer. Doc link - https://platform.openai.com/docs/api-reference/fine-tuning/object